src/script/miniscript.h

```diff
@@ -0,0 +1,992 @@
+ *     -VERIFY version of the last opcode).
+ *
+ * For each of these properties the subset rule holds: an expression with properties X, Y, and Z, is also
+ * valid in places where an X, a Y, a Z, an XY, ... is expected.
+*/
+class Type {
+    //! Internal bitmap of properties (see ""_mst operator for details).
+    uint16_t m_flags;
+
+    //! Internal constructed used by the ""_mst operator.
+    explicit constexpr Type(uint16_t flags) : m_flags(flags) {}
+
+public:
+    //! The only way to publicly construct a Type is using this literal operator.
+    friend constexpr Type operator"" _mst(const char* c, size_t l);
+
+    //! Compute the type with the union of properties.
+    constexpr Type operator|(Type x) const { return Type(m_flags | x.m_flags); }
+
+    //! Compute the type with the intersection of properties.
+    constexpr Type operator&(Type x) const { return Type(m_flags & x.m_flags); }
+
+    //! Check whether the left hand's properties are superset of the right's (= left is a subtype of right).
+    constexpr bool operator<<(Type x) const { return (x.m_flags & ~m_flags) == 0; }
+
+    //! Comparison operator to enable use in sets/maps (total ordering incompatible with <<).
+    constexpr bool operator<(Type x) const { return m_flags < x.m_flags; }
+
+    //! Equality operator.
+    constexpr bool operator==(Type x) const { return m_flags == x.m_flags; }
+
+    //! The empty type if x is false, itself otherwise.
+    constexpr Type If(bool x) const { return Type(x ? m_flags : 0); }
+};
+
+//! Literal operator to construct Type objects.
+inline constexpr Type operator""_mst(const char* c, size_t l) {
+    return l == 0 ? Type(0) : operator"" _mst(c + 1, l - 1) | Type(
+        *c == 'B' ? 1 << 0 : // Base type
+        *c == 'V' ? 1 << 1 : // Verify type
+        *c == 'K' ? 1 << 2 : // Key type
+        *c == 'W' ? 1 << 3 : // Wrapped type
+        *c == 'z' ? 1 << 4 : // Zero-arg property
+        *c == 'o' ? 1 << 5 : // One-arg property
+        *c == 'n' ? 1 << 6 : // Nonzero arg property
+        *c == 'd' ? 1 << 7 : // Dissatisfiable property
+        *c == 'u' ? 1 << 8 : // Unit property
+        *c == 'e' ? 1 << 9 : // Expression property
+        *c == 'f' ? 1 << 10 : // Forced property
+        *c == 's' ? 1 << 11 : // Safe property
+        *c == 'm' ? 1 << 12 : // Nonmalleable property
+        *c == 'x' ? 1 << 13 : // Expensive verify
+        (throw std::logic_error("Unknown character in _mst literal"), 0)
+    );
+}
+
+template<typename Key> struct Node;
+template<typename Key> using NodeRef = std::shared_ptr<const Node<Key>>;
+
+//! Construct a miniscript node as a shared_ptr.
+template<typename Key, typename... Args>
+NodeRef<Key> MakeNodeRef(Args&&... args) { return std::make_shared<const Node<Key>>(std::forward<Args>(args)...); }
+
+//! The different node types in miniscript.
+enum class NodeType {
+    FALSE,     //!< OP_0
+    TRUE,      //!< OP_1
```

Comment:
Looks like these already are macros on mac from `/home/travis/build/bitcoin/bitcoin/depends/SDKs/MacOSX10.11.sdk/usr/include/mach/boolean.h`?

And also in windows: https://travis-ci.org/bitcoin/bitcoin/jobs/580527642#L3384

Reply:
Fixed by renaming to `JUST_0` and `JUST_1`.